go/controller/internal/httpserver/handlers/modelconfig.go

```diff
@@ -60,10 +137,499 @@ func (h *ModelConfigHandler) HandleGetModelConfig(w ErrorResponseWriter, r *http
+	if err := h.KubeClient.Get(r.Context(), types.NamespacedName{
+		Name:      configName,
+		Namespace: common.GetResourceNamespace(),
+	}, modelConfig); err != nil {
+		if k8serrors.IsNotFound(err) {
+			log.Info("Model config not found")
+			w.RespondWithError(errors.NewNotFoundError("Model config not found", nil))
+			return
+		}
+		log.Error(err, "Failed to get model config")
+		w.RespondWithError(errors.NewInternalServerError("Failed to get model config", err))
+		return
+	}
+
+	// --- Update Secret if API Key is provided (and not Ollama) ---
+	secretName := configName
+	secretKey := fmt.Sprintf("%s_API_KEY", strings.ToUpper(req.Provider.Type))
+	isOllama := strings.EqualFold(req.Provider.Type, string(v1alpha1.Ollama))
+	shouldUpdateSecret := req.APIKey != nil && *req.APIKey != ""
+
+	if !isOllama && shouldUpdateSecret {
+		log.V(1).Info("Updating API key secret", "secretName", secretName, "secretKey", secretKey)
+		existingSecret := &corev1.Secret{}
+		err = h.KubeClient.Get(r.Context(), types.NamespacedName{Name: secretName, Namespace: common.GetResourceNamespace()}, existingSecret)
+		if err != nil && !k8serrors.IsNotFound(err) {
+			log.Error(err, "Failed to get existing secret for update")
+			w.RespondWithError(errors.NewInternalServerError("Failed to get API key secret", err))
+			return
+		}
+
+		if k8serrors.IsNotFound(err) {
+			// Secret doesn't exist, create it (edge case, should normally exist)
+			log.Info("Secret not found for update, creating new one", "secretName", secretName)
+			secret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{Name: secretName, Namespace: common.GetResourceNamespace()},
+				StringData: map[string]string{secretKey: *req.APIKey},
+			}
+			if err := h.KubeClient.Create(r.Context(), secret); err != nil {
+				log.Error(err, "Failed to create new API key secret during update")
+				w.RespondWithError(errors.NewInternalServerError("Failed to create API key secret", err))
+				return
+			}
+		} else {
+			// Secret exists, update it
+			if existingSecret.StringData == nil {
+				existingSecret.StringData = make(map[string]string)
+			}
+			existingSecret.StringData[secretKey] = *req.APIKey
+			if err := h.KubeClient.Update(r.Context(), existingSecret); err != nil {
+				log.Error(err, "Failed to update API key secret")
+				w.RespondWithError(errors.NewInternalServerError("Failed to update API key secret", err))
+				return
+			}
+		}
+		log.V(1).Info("Successfully updated API key secret")
+	} else if isOllama {
+		log.V(1).Info("Skipping secret update for Ollama provider")
+	} else {
+		log.V(1).Info("API key not provided, secret will not be updated")
+	}
+
+	log.V(1).Info("Updating ModelConfig spec")
+	modelConfig.Spec.OpenAI = nil
+	modelConfig.Spec.Anthropic = nil
+	modelConfig.Spec.AzureOpenAI = nil
+	modelConfig.Spec.Ollama = nil
+
+	modelConfig.Spec.Model = req.Model
+	modelConfig.Spec.Provider = v1alpha1.ModelProvider(req.Provider.Name)
+	modelConfig.Spec.APIKeySecretKey = secretKey
```

Comment:
Rather than doing this, can you just set `ModelConfig.Spec = ModelConfigSpec{}`. Are there any fields that need to stay the same?